python/test/conftest.py

```diff
@@ -136,3 +138,56 @@ def tempdir(request):
 
     """
     return _create_tempdir(request)
+
+
+@pytest.fixture(scope="function")
+def cg_solver():
+    """Simple Conjugate Gradient solver for SPD problems,
+       which can work in serial or parallel for testing use.
+       Not suitable for large problems."""
+
+    # Basic Conjugate Gradient solver
+    def _cg(comm, A, b, x):
+        kmax = 500
+        rtol2 = 10 * np.finfo(x.array.dtype).eps
+
+        def _global_dot(comm, v0, v1):
+            return comm.allreduce(np.dot(v0, v1), MPI.SUM)
```

Comment:
Doesn't look like the correct inner product for the complex-valued case.

Reply:
replacing with `vdot`